### PR TITLE
ci: shard all integration tests

### PR DESCRIPTION
**Pull request by gregmagolan**

### CODE REVIEW

File: scripts/ci/run_integration_tests.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+set -eu -o pipefail
+# -e: exits if a command fails
+# -u: errors if an variable is referenced before being set
+# -o pipefail: causes a pipeline to produce a failure return code if any command errors
+
+readonly TEST_TARGETS=$(bazel query 'kind("bazel_integration_test", //examples/... union //e2e/...)')
```
Comment (alexeagle) [inline scripts/ci/run_integration_tests.sh:8]: Switching in this PR from selecting based on tags to using the rule kind. Did you check that it gives the same list of targets?

Comment (alexeagle) [reply scripts/ci/run_integration_tests.sh:8]: also we rely on these coming in the same order every time, might be good to comment on that, or maybe even do an explicit sort to be safe?

Comment (gregmagolan) [reply scripts/ci/run_integration_tests.sh:8]: yes. double checked that everything is still tested.

I'm making the assumption that bazel query is deterministic in its order. I'll add an explicit sort to be safe.

**Comment by gregmagolan** [comment]
Before:
Screen Shot 2020-04-01 at 1 43 56 PM

**Comment by gregmagolan** [comment]
After:
Screen Shot 2020-04-01 at 2 03 30 PM

**Comment by gregmagolan** [comment]
Sharded with parallelism 8:
Screen Shot 2020-04-01 at 2 07 02 PM

**Pull request merged**